clang/lib/AST/DeclCXX.cpp

```diff
@@ -561,6 +561,42 @@ void CXXRecordDecl::addedClassSubobject(CXXRecordDecl *Subobj) {
     data().StructuralIfLiteral = false;
 }
 
+const CXXRecordDecl *CXXRecordDecl::getStandardLayoutBaseWithFields() const {
+#ifndef NDEBUG
+  {
+    assert(
+        isStandardLayout() &&
+        "getStandardLayoutBaseWithFields called on a non-standard-layout type");
```

Comment:
I think this assert should be on the default code path, not just for expensive checks.